scripts/set_instrument_list.py

```diff
@@ -43,19 +43,30 @@ def set_env():
     print epics_ca_addr_list + " = " + str(os.environ.get(epics_ca_addr_list))
 
 
-def inst_dictionary(instrument_name, hostname_prefix="NDX"):
+def inst_dictionary(instrument_name, hostname_prefix="NDX", hostname=None, pv_prefix=None):
     """
     Generate the instrument dictionary for the instrument list
     Args:
         instrument_name: instrument name
         hostname_prefix: prefix for hostname (defaults to NDX)
+        hostname: whole host name overrides prefix, defaults to hostname_prefix + instrument name
+        pv_prefix: the pv prefeix; default to IN:instrument_name
 
     Returns: dictionary for instrument
 
     """
+    if hostname is not None:
+        hostname_to_use = hostname
+    else:
+        hostname_to_use = hostname_prefix + instrument_name
+    if pv_prefix is not None:
+        pv_prefix_to_use = pv_prefix
+    else:
+        pv_prefix_to_use = "IN:{0}:".format(instrument_name)
     return {"name": instrument_name,
-            "hostName": hostname_prefix + instrument_name,
-            "pvPrefix": "IN:{0}:".format(instrument_name)}
+            "hostName": hostname_to_use,
+            "pv_prefix": pv_prefix_to_use}
```

Comment:
@John-Holt-Tessella The change from `pvPrefix` to `pvprefix` broke the configchecker. What was the reason for this?